In [7]:
import numpy as np
import pandas as pd
import os
for dirname,_,filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname,filename))

In [8]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression 
from sklearn.metrics import accuracy_score
import warnings
warnings.filterwarnings('ignore')

In [11]:
data = pd.read_csv('C:/Users/Dell/Downloads/heart.csv')

In [12]:
data.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trtbps    303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalachh  303 non-null    int64  
 8   exng      303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slp       303 non-null    int64  
 11  caa       303 non-null    int64  
 12  thall     303 non-null    int64  
 13  output    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


In [13]:
data.isnull().sum()

age         0
sex         0
cp          0
trtbps      0
chol        0
fbs         0
restecg     0
thalachh    0
exng        0
oldpeak     0
slp         0
caa         0
thall       0
output      0
dtype: int64

from the above two outputs, all the attributes have non-null values and isnull() is showing 0. so there is no missing data in the dataset. so no need for null data management.

In [14]:
data.describe().columns

Index(['age', 'sex', 'cp', 'trtbps', 'chol', 'fbs', 'restecg', 'thalachh',
       'exng', 'oldpeak', 'slp', 'caa', 'thall', 'output'],
      dtype='object')

the above output is the attributes we have taken in the dataset

In [15]:
# checking output variable
data['output'].value_counts()

1    165
0    138
Name: output, dtype: int64

as per the above output..
165 people among 303 people have chance to get heart disease and
138 people have healthy heart.
now we have to process the data.

In [16]:
# now remove the output and store it in somewhere else
X = data.drop(columns='output',axis=1)
Y = data['output']
print(X)

     age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  
0      0      1  
1      0      2

previously we have 14 columns, but now as we removed output attribute and stored it in Y. It is showing as 13 columns.

In [17]:
print(Y)

0      1
1      1
2      1
3      1
4      1
      ..
298    0
299    0
300    0
301    0
302    0
Name: output, Length: 303, dtype: int64


In [18]:
# Now we are going to split data into training ans test data
X_train, X_test, Y_train, Y_test = train_test_split(X,Y,test_size=0.2,stratify=Y,random_state=2)
print(X.shape, X_train.shape, X_test.shape)

(303, 13) (242, 13) (61, 13)


in the above output, the previous 303 values are splitted into 242 and 61 train and test data

In [19]:
model = LogisticRegression()  #logistic regression model
# model training with LOGISTIC REGRESSION algorithm
model.fit(X_train, Y_train) #finds a relationship 

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [20]:
# now check the accuracy of training data
X_train_prediction = model.predict(X_train)
training_data_accuracy = accuracy_score(X_train_prediction, Y_train)
print("Accuracy on Training data is ",training_data_accuracy)

Accuracy on Training data is  0.8512396694214877


In [21]:
# now check the accuracy of test data
X_test_prediction = model.predict(X_test)
test_data_accuracy = accuracy_score(X_test_prediction, Y_test)
print("Accuracy on Training data is ",test_data_accuracy)

Accuracy on Training data is  0.819672131147541


accuracy of training and testing data are nearly same. so there is no overfitting

In [27]:
# now prediction for specific input values 
input_data = (57,1,2,128,229,0,0,150,0,0.4,1,1,3)
input_data_numpy = np.asarray(input_data)
input_data_reshaped = input_data_numpy.reshape(1,-1)
prediction = model.predict(input_data_reshaped)
print(prediction)
if prediction[0]==0:
    print("The person have heart disease")
else:
    print("The person deos not have heart disease")

[0]
The person have heart disease


In [28]:
# now prediction for specific input values 
input_data = (57,0,0,120,354,0,1,163,1,0.6,2,0,2)
input_data_numpy = np.asarray(input_data)
input_data_reshaped = input_data_numpy.reshape(1,-1)
prediction = model.predict(input_data_reshaped)
print(prediction)
if prediction[0]==0:
    print("The person have heart disease")
else:
    print("The person deos not have heart disease")

[1]
The person deos not have heart disease
